### PR TITLE
feat: hamburger menu component

### PR DESCRIPTION
**Pull request by kibo-chandradeeptalaha**

### CODE REVIEW

File: components/layout/HamburgerMenu/HamburgerMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                mobileIconColor="black"
+                iconFontSize="large"
+                showTitleInMobile={true}
+              />
+            </Box>
+          </CategoryNestedNavigation>
+        </Box>
+        <Box height={19} sx={{ backgroundColor: 'grey.300' }}></Box>
+        <List sx={{ width: '100%', maxHeight: '40%', overflowY: 'auto' }}>
+          {['Order Status', 'Wishlist', 'Nav Link 3', 'Nav Link 4'].map((text) => (
```
Comment (kibo-kevinwatts) [inline components/layout/HamburgerMenu/HamburgerMenu.tsx:69]: Allow these to be props, an array of items  `{ text: string, link: string }`

File: components/common/HeaderAction/index.tsx [hunk 2]
```diff
@@ -1,15 +1,17 @@
 import type { MouseEventHandler } from 'react'
```
Comment (kibo-kevinwatts) [inline components/common/HeaderAction/index.tsx:1]: File name should stay same as component name, add a separate index file to export it if needed

File: components/layout/CategoryNestedNavigation/CategoryNestedNavigation.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    categoryCode?: string | null
+    backLink?: string
+  }>(initialSubHeader)
+
+  const reset = () => {
+    setActiveCategory(allCategories)
+    setSubHeader(initialSubHeader)
+  }
+
+  const findParent = (
```
Comment (kibo-kevinwatts) [inline components/layout/CategoryNestedNavigation/CategoryNestedNavigation.tsx:73]: this should be moved to a util method, other components might need this too

File: components/layout/HamburgerMenu/HamburgerMenu.tsx (hunk 1 above)
Comment (kibo-chandradeeptalaha) [reply components/layout/HamburgerMenu/HamburgerMenu.tsx]: Done

File: components/common/HeaderAction/index.tsx (hunk 2 above)
Comment (kibo-chandradeeptalaha) [reply components/common/HeaderAction/index.tsx]: Done

File: public/locales/en/common.json [hunk 4]
```diff
@@ -52,5 +52,8 @@
   "qty": "Qty",
   "details": "Details",
   "ship": "Ship to Home",
-  "pickup": "Pickup in Store "
+  "pickup": "Pickup in Store ",
+  "apply": "Apply",
+  "promoInputPlaceholder": "Enter promo code",
+  "allDepartments": "All Departments"
```
[the remark this answers is not in the document]
Comment (kibo-chandradeeptalaha) [reply public/locales/en/common.json]: Done

File: components/layout/CategoryNestedNavigation/CategoryNestedNavigation.tsx (hunk 3 above)
Comment (kibo-chandradeeptalaha) [reply components/layout/CategoryNestedNavigation/CategoryNestedNavigation.tsx]: Moved with test case

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**